tests/model_fields/tests.py

```diff
@@ -233,6 +233,36 @@ def test_datetimes_save_completely(self):
         self.assertEqual(obj.dt, datetim)
         self.assertEqual(obj.t, tim)
 
+    def test_datefield_to_python_issue_21523(self):
+        """DateField.to_python should handle mock dates. (see #21523)"""
+
+        class FakeDate(datetime.date):
+            """Mock date class."""
+            pass
+
+        with test.mock.patch('datetime.date', FakeDate):
+            # this will create a mock date
+            today = datetime.date.today()
+            # this will force the creation of a 'real' date (i.e. not the mock)
+            tomorrow = today + datetime.timedelta(days=1)
+            # to_python does an isinstance(value, datetime.date) check -
+            # when we are mocking out datetime.date with our FakeDate class
+            # then this test will FAIL if value is a real date object, which
+            # is the case with issue #21523.
+            #
+            # NB use of assertTrue and isinstance rather than assertIsInstance
```

Comment:
I understand, and I'm sure it's being over-cautious, but the intention of the patch isn't to assert that 'today is an instance of datetime.date', it's to assert that `isinstance(today, datetime.date)` is True. It's semantics, but it's an assumption like this (that `assertIsInstance` will never change it's internal implementation) that caused the issue in the first place.